packages/flutter/lib/src/widgets/overflow_bar.dart

```diff
@@ -510,17 +510,20 @@ class _RenderOverflowBar extends RenderBox
       }
       size = constraints.constrain(Size(constraints.maxWidth, y - overflowSpacing));
     } else {
-      // Default horizontal layout
-      child = rtl ? lastChild : firstChild;
-      RenderBox? nextChild() => rtl ? childBefore(child!) : childAfter(child!);
-      double x  = 0;
+      size = constraints.constrain(Size(actualWidth, maxChildHeight));
+      child = firstChild;
+      double x = rtl ? size.width - child!.size.width : 0;
       while (child != null) {
         final _OverflowBarParentData childParentData = child.parentData! as _OverflowBarParentData;
         childParentData.offset = Offset(x, (maxChildHeight - child.size.height) / 2);
-        x += child.size.width + spacing;
-        child = nextChild();
+        if (!rtl) {
+          x += child.size.width + spacing;
+        }
+        child = childAfter(child);
+        if (rtl && child != null) {
```

Comment:
This is a little confusing why the rtl offset is handled after getting the next child. Maybe a comment about why this is done this way might be helpful.

Or perhaps start x at the width and compensate in the offset by the width and spacing if it is rtl? Probably not that critical, but it did take me a few minutes to grok this.

Reply:
It's tricky because the loop is set up with the initial x offset at the origin of the first child. To advance the x offset to the next child for LTR, you add the width of the _current_ child.  To advance the x offset to the next child for RTL, you subtract the offset of the _next_ child (if there is one). I will add a comment that explains what's going on.